hibernate-core/src/main/java/org/hibernate/query/hql/internal/BasicDotIdentifierConsumer.java

```diff
@@ -184,7 +187,12 @@ public SemanticPathPart resolvePathPart(
 			if ( importableName != null ) {
 				final EntityDomainType<?> entityDomainType = jpaMetamodel.entity( importableName );
 				if ( entityDomainType != null ) {
-					return new SqmLiteralEntityType( entityDomainType, nodeBuilder );
+					return new SqmLiteralEntityType<>( entityDomainType, nodeBuilder );
+				}
+
+				final EmbeddableDomainType<?> embeddableDomainType = jpaMetamodel.embeddable( importableName );
```

Comment:
This actually caused envers tests to fail: apparently the `managedType()` method doesn't return "_AUD" entity types created by envers itself if the `hibernate.jpa.metamodel.population` is on `IGNORE_UNSUPPORTED`, which is the default.

I've left a call to `JpaMetamodel#entity( importableName )` to fallback on. I know it's a double map access but I don't think there are alternatives here.

Reply:
This is a more general problem IMO, because the `jpaEntityTypeMap` might also contain elements that are "unsupported", so iteration through entities via `jakarta.persistence.metamodel.Metamodel#getEntities` will expose such "unsupported" `EntityType` objects.

I think it would be best to have two maps in `JpaMetamodelImpl`, one for types by class and the other for types by type name and store a boolean that specifies whether unsupported elements should be returned or not.

Reply:
I have implemented this, we now only have 2 maps in `JpaMetamodelImpl`.